workflow/util.rb

```diff
@@ -1,77 +1,76 @@
+    hpxml = hpxml_obj_or_path
+  else
+    fail 'Unexpected argument.'
+  end
 
-      # Floor area in basements with at least half of the gross surface area of the basement’s exterior walls below grade shall not be counted
-      cond_bsmt_floor_area = 0.0
-      cond_bsmt_ext_wall_area = 0.0
-      cond_bsmt_ext_wall_area_bg = 0.0
-      hpxml.slabs.each do |slab|
-        next unless slab.interior_adjacent_to == HPXML::LocationBasementConditioned
+  # Floor area in basements with at least half of the gross surface area of the basement’s exterior walls below grade shall not be counted
+  cond_bsmt_floor_area = 0.0
+  cond_bsmt_ext_wall_area = 0.0
+  cond_bsmt_ext_wall_area_bg = 0.0
+  hpxml.slabs.each do |slab|
+    next unless slab.interior_adjacent_to == HPXML::LocationBasementConditioned
 
-        cond_bsmt_floor_area += slab.area
-      end
-      hpxml.foundation_walls.each do |fwall|
-        next unless fwall.is_exterior
-        next unless fwall.interior_adjacent_to == HPXML::LocationBasementConditioned
+    cond_bsmt_floor_area += slab.area
+  end
+  hpxml.foundation_walls.each do |fwall|
+    next unless fwall.is_exterior
+    next unless fwall.interior_adjacent_to == HPXML::LocationBasementConditioned
 
-        cond_bsmt_ext_wall_area += fwall.area
-        cond_bsmt_ext_wall_area_bg += (fwall.area * fwall.depth_below_grade / fwall.height)
-      end
-      hpxml.walls.each do |wall|
-        next unless wall.is_exterior
-        next unless wall.interior_adjacent_to == HPXML::LocationBasementConditioned
+    cond_bsmt_ext_wall_area += fwall.area
+    cond_bsmt_ext_wall_area_bg += (fwall.area * fwall.depth_below_grade / fwall.height)
+  end
+  hpxml.walls.each do |wall|
+    next unless wall.is_exterior
+    next unless wall.interior_adjacent_to == HPXML::LocationBasementConditioned
 
-        cond_bsmt_ext_wall_area += wall.area
-      end
-      if cond_bsmt_ext_wall_area_bg / cond_bsmt_ext_wall_area > 0.5
-        cfa -= cond_bsmt_floor_area
-      end
+    cond_bsmt_ext_wall_area += wall.area
+  end
+  if cond_bsmt_ext_wall_area_bg / cond_bsmt_ext_wall_area > 0.5
+    cfa -= cond_bsmt_floor_area
+  end
 
-      cfa_benchmark = [600 * nbr + 400, 1000].max
-      saf = (cfa_benchmark / cfa)**0.25
-      if saf > 1.0 # SAF not to exceed 1.0
-        saf = 1.0
-      end
-      return saf # Size Adjustment Factor
-    elsif [HPXML::ResidentialTypeApartment].include? results[:rated_facility_type]
-      # For condos and apartments in multi-family buildings the SAF shall always equal 1.0.
-      return 1.0
-    end
-  else
-    # SAF does not apply
-    return 1.0
+  cfa_benchmark = [600 * nbr + 400, 1000].max
+  saf = (cfa_benchmark / cfa)**0.25
+  if saf > 1.0 # SAF not to exceed 1.0
+    saf = 1.0
   end
+  return saf # Size Adjustment Factor
 end
 
-def calc_opp_eri_limit(esrd_eri, saf, es_version)
+def calc_opp_eri_limit(rd_eri, saf, program_version)
   # Calculates the limit, in ERI points, for On-site Power Production per
```

Comment:
ZERH is currently going through the `else`, which means it is being treated the same as ENERGY STAR 3.1 and above. Did you mean to add it to the `if` so that it follows ENERGY STAR 3.0?

That said, I'm reading the ZERH spec and having trouble determining what we should actually be doing. I will ask DOE for clarification.

[ENERGY STAR 3.0](https://www.energystar.gov/sites/default/files/asset/document/National%20Program%20Requirements%20Version%203_Rev%2012.pdf):

> ... on-site power generation may only be used to meet the ENERGY STAR ERI Target for homes that are larger than the Benchmark Home and only for the incremental change in the ENERGY STAR ERI Target caused by the Size Adjustment Factor.

[ENERGY STAR 3.1](https://www.energystar.gov/sites/default/files/asset/document/National%20Program%20Requirements%20Version%203.1_Rev%2012.pdf):

> ... on-site power generation may not be used to meet the ENERGY STAR ERI Target.

ZERH v1:

> On-site power generation may not be used to qualify a home for the DOE Zero Energy Ready Home Target Home requirements, but can be used to achieve additional HERS Index Score reductions needed for homes larger than the Benchmark Home.

Reply:
As per Jamie, ZERH V1 treats OPP the same way as ES V3.0.